[MAJOR][Synthetics] Remove deprecated `startUrl` native URL variables support (#687)
DataDog/datadog-ci: src/commands/synthetics/README.md

- `retry`: (object) retry policy for the test.
  - `count`: (integer) number of attempts to perform in case of test failure.
  - `interval`: (integer) interval between the attempts (in milliseconds).

> - `startUrl`: (string) new start URL to provide to the test. Variables specified in brackets (`{{ EXAMPLE }}`) found in environment variables will be replaced.

Reviewer: ```suggestion
- `startUrl`: (string) new start URL to provide to the test. Variables specified in brackets (`{{ EXAMPLE }}`) found in environment variables are replaced.
```

| `HOSTNAME`           | Test's hostname              | `www.example.org`                                             |
| `ORIGIN`             | Test's origin                | `https://www.example.org:81`                                  |
| `PARAMS`             | Test's query parameters      | `?abc=123`                                                    |
| `PATHNAME`           | Test's URl path              | `/path/to/something`                                          |
| `PORT`               | Test's host port             | `81`                                                          |
| `PROTOCOL`           | Test's protocol              | `https:`                                                      |
| `SUBDOMAIN`          | Test's sub domain            | `www`                                                         |

For instance, if your test's starting url is `https://www.example.org:81/path/to/something?abc=123#target`

It can be written as :

- `{{PROTOCOL}}//{{SUBDOMAIN}}.{{DOMAIN}}:{{PORT}}{{PATHNAME}}{{PARAMS}}{{HASH}}`
- `{{PROTOCOL}}//{{HOST}}{{PATHNAME}}{{PARAMS}}{{HASH}}`
- `{{URL}}`

and so on...

</details>

> 

Author: 👋  @DataDog/documentation @peculiarism Hey, we're removing the support for URL variables which has been deprecated for a few versions and reported with warnings  during the command execution.
I see we still document it on the [continuous testing configuration doc page](https://docs.datadoghq.com/continuous_testing/cicd_integrations/configuration?tab=npm#starting-url), we should update it with `startUrlSubstitutionRegex` or usage of environment variables.

Reviewer 2: I'll raise a PR to remove this from our docs, thanks for the heads up, @lefebvree!
